From 034259afd89c3af6685fbc8fa2319e785f33166f Mon Sep 17 00:00:00 2001
From: Ben Brown <ben@demerara.io>
Date: Sun, 17 Apr 2022 14:23:18 +0100
Subject: [PATCH] configure: conditionally check for user/group manipulation
 commands

Only check for useradd/adduser and groupadd/addgroup when
systemd_sysusers is not present on the system.

Signed-off-by: Ben Brown <ben@demerara.io>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6482944f..b3f11f3e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -489,7 +489,7 @@ AC_CHECK_PROG(systemd_tmpfiles, systemd-tmpfiles, yes)
 AM_CONDITIONAL(SYSD_TMPFILES, test "x$systemd_tmpfiles" = "xyes")
 
 # Check all tools used by make install
-AS_IF([test "$HOSTOS" = "Linux"],
+AS_IF([test "$HOSTOS" = "Linux" && test "x$systemd_sysusers" != "xyes"],
     [ AC_CHECK_PROG(useradd, useradd, yes)
       AC_CHECK_PROG(groupadd, groupadd, yes)
       AC_CHECK_PROG(adduser, adduser, yes)
-- 
2.35.3

